### ISSUE TITLE
Benchmark comparison against GPUs, FPGAs

### ISSUE THREAD

**Issue by LifeIsStrange** [GraphIt-DSL/graphit#82]
First of all Thank you for this impressive software !!
But I won't use it until i've found a benhcmark comparison against https://developer.nvidia.com/nvgraph on a modern gpu.
A comparison against AMD gpu would be Nice too https://www.google.com/search?q=AMD+graph+algorithms+opencl&client=firefox-b&oq=AMD+graph+algorithms+opencl&gs_l=mobile-heirloom-serp.3..30i10.17126.20484.0.20921.13.12.1.0.0.1.480.2284.4j5j0j2j1.12.0....0...1c.1j4.34.mobile-heirloom-serp..7.6.593.WaazgBmCg_E
I had read that FPGAs are the fastest at graph analytics until graph processors become a reality (aren't they already ? https://www.google.com/search?q=graph+processor&client=firefox-b&oq=graph+processor&gs_l=mobile-heirloom-serp.3...5156.5156.0.5686.1.1.0.0.0.0.0.0..0.0....0...1c.1j4.34.mobile-heirloom-serp..1.0.0.QKVdUYR0fFo but I understanding that it is difficult to find such accelerators for testing)

And if GPUs or HSA are faster than cpus for graph analytics, you could think about the possibility of porting the graphit to gpu backends, i've heard it became  « relatively » easy to make since gpu llvm backends are mature and SPIR-V is a thing.

**Comment by yunmingzhang17** [comment]
Hi, 

Thanks for your interest in the project! We are keenly aware of the other backends with greater parallelism and memory bandwidth, such as GPUs and FPGAs. In fact, we are currently working on a GPU backend.

However, we believe there is still significant value in high-performance CPU graph analytics. Many large graphs don't fit in the memory of a single GPU, communication latency / bandwidth across multiple GPUs can still be an issue, moving large amounts of data between DRAM and GPU memory takes non-trivial time, and the computation patterns of some graph applications that involve lots of control flows or less degrees of parallelism are not easily mapped to massively parallel architectures. 

In summary, we agree that GPU / FPGA / Graph Processor can be the better platform for some applications, and we are actively working on it. But in some cases, efficient CPU graph analytics can still be very valuable.

**Comment by yunmingzhang17** [comment]
Thanks for the note! 

We are also looking into a heterogeneous approach. It is just a bit tricky to utilize both CPU and GPU, while ensuring there's no significant communication overhead. 

Some existing work has also looked at using CPU and GPU together, 
https://www.usenix.org/conference/atc17/technical-sessions/presentation/ma

Best

**Comment by LifeIsStrange** [comment]
Happy to hear that !

Just to say that native Linux now supports heterogeneous computing with https://www.kernel.org/doc/html/v4.18/vm/hmm.html

**Comment by yunmingzhang17** [comment]
We actually had a paper (Ajay is the first author) published on benchmarking against GPU, https://ieeexplore.ieee.org/abstract/document/9370321. Feel free to check it out.
